etl/steps/data/garden/regions/2023-01-01/migration.py

```diff
@@ -0,0 +1,884 @@
+"""Migrate countries-regions data from the garden/reference dataset to a new dataset.
+
+This script needs to be used only once before creating the regions dataset for the first time.
+It will generate:
+* A yaml file of region definitions.
+* A csv file of additional region codes.
+
+Once those files exist, subsequent versions of the dataset should simply duplicate those yaml and csv files, and make
+manual modifications and additions to those files.
+
+"""
+
+import json
+from pathlib import Path
+
+import pandas as pd
+from owid import catalog
+
+from etl.paths import DATA_DIR
+
+# Path to current folder.
+CURRENT_DIR = Path(__file__).parent
+
+# Paths to output files.
+REGION_DEFINITIONS_OUTPUT_FILE = CURRENT_DIR / "regions.yml"
+REGION_CODES_OUTPUT_FILE = CURRENT_DIR / "regions.codes.csv"
+
+# Possible region types considered.
+REGION_TYPES = ["country", "continent", "aggregate", "other"]
+
+# Manually list all inhabited continents and their member countries.
+# This is not a trivial task, since there are many regions that are unclear whether they are independent or whether
+# they should already be counted within another country.
+# This list is mainlly used to create aggregates (i.e. to add up data from different countries to obtain data for
+# continents).
+continent_members = {
+    "Africa": [
+        "Algeria",
+        "Angola",
+        "Benin",
+        "Botswana",
+        "Burkina Faso",
+        "Burundi",
+        "Cameroon",
+        "Cape Verde",
+        "Central African Republic",
+        "Chad",
+        "Comoros",
+        "Congo",
+        "Cote d'Ivoire",
+        "Democratic Republic of Congo",
+        "Djibouti",
+        "Egypt",
+        "Equatorial Guinea",
+        "Eritrea",
+        "Eritrea and Ethiopia",
```

Comment:
Does this refer to Ethiopia before Eritrea split from it? I thought it was more commonly known as simply 'Ethiopia'.

Reply:
Yes, I believe that's what it refers to. I didn't create this entity, but I suppose it was called `Eritrea and Ethiopia` to avoid confusion with `Ethiopia` current and former. But we could redefine `Eritrea and Ethiopia` -> `Ethiopia (former)`. Let me know if you think that's a better choice.

Reply:
Yes, if we want a different name for when Eritrea was part of Ethiopia, I would prefer Ethiopia (former). That would make clear that it was indeed one country.